core-common/src/main/java/eu/arrowhead/common/processor/NetworkAddressDetector.java

```diff
@@ -0,0 +1,212 @@
+			
+			remoteAddr = networkAddressPreProcessor.normalize(servletRequest.getLocalAddr());
+		}
+		
+		if (Utilities.isEmpty(remoteAddr) || filterProxyAddressSet.contains(remoteAddr)) {
+			return false;
+		} else {
+			result.setDetectedAddress(remoteAddr);
+			result.setDetectionSuccess(true);
+			return true;
+		}		
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private boolean detectByHeaderForwarded(final HttpServletRequest servletRequest, final AddressDetectionResult result) {
+		//https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded
+		//https://tools.ietf.org/html/rfc7230#section-3.2
+		final List<String> headerValues = Collections.list(servletRequest.getHeaders(HEADER_FORWARDED));
+		for (int i = headerValues.size() - 1; i >= 0; i--) {
+			
+			String[] subValues = new String[1];
+			if (headerValues.get(i).contains(SEMI_COLON)) {
+				subValues = headerValues.get(i).split(SEMI_COLON);
+			} else if (headerValues.get(i).contains(COMMA)) {
+				subValues = headerValues.get(i).split(COMMA);
+			} else {
+				subValues[0] = headerValues.get(i);
+			}
+			
+			for (int j = subValues.length - 1; j >= 0; j--) {
+				final String[] pair = subValues[j].split(EQUAL_SIGN);
+				if (pair[0].toLowerCase().trim().startsWith("for")) {
+					if (Utilities.isEmpty(pair[1])) {
+						return false;
+					}
+					
+					final String address = processAddress(pair[1]);
+					if (!filterProxyAddressSet.contains(address)) {
+						result.setDetectionSuccess(true);
+						result.setDetectedAddress(address);
+						return true;
+					}
+				}
+			}
+		}
+		
+		return false;
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	private boolean detectByHeaderXForwardedFor(final HttpServletRequest servletRequest, final AddressDetectionResult result) {
+		//https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For
+		final List<String> headerValues = Collections.list(servletRequest.getHeaders(HEADER_X_FORWARDED_FOR));
+		for (int i = headerValues.size() - 1; i >= 0; i--) {
+			final String[] subValues = headerValues.get(i).split(COMMA);
+			for (int j = subValues.length - 1; j >= 0; j--) {
+				if (Utilities.isEmpty(subValues[j])) {
+					return false;
+				}
+				
+				final String address = processAddress(subValues[j]);
+				if (!filterProxyAddressSet.contains(address)) {
+					result.setDetectionSuccess(true);
+					result.setDetectedAddress(address);
+					return true;
+				}				
+			}
+		}
+		
+		return false;
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	private String processAddress(final String value) {
+		final StringBuilder sb = new StringBuilder();
+		int dotCtn = 0;
+		final List<Integer> colonIdxList = new ArrayList<>();
+		for (int i = 0; i < value.length(); i++) {
+			final String ch = String.valueOf(value.charAt(i));
```

Comment:
if you inspect characters, why don't you use char type?

Reply:
I'm not sure if it would work with escaped characters like `"\""`

Reply:
it should work with the character `'"'` (in a constant of course)

Reply:
ok